lib/internal/test_runner/test.js

```diff
@@ -225,11 +230,30 @@ class Test extends AsyncResource {
     }
 
     if (testNamePatterns !== null) {
-      // eslint-disable-next-line no-use-before-define
-      const match = this instanceof TestHook || ArrayPrototypeSome(
-        testNamePatterns,
-        (re) => RegExpPrototypeExec(re, name) !== null,
-      );
+      const match =
+        // eslint-disable-next-line no-use-before-define
+        this instanceof TestHook ||
+        ArrayPrototypeSome(testNamePatterns, (re) => {
+          const scopedTestNamePatterns = StringPrototype.split.call(re, '>');
+          if (name !== '<root>' && scopedTestNamePatterns.length > 1) {
+            const patterns = ArrayPrototypeMap(
+              scopedTestNamePatterns,
+              (pattern) => {
+                const text = StringPrototypeReplace(pattern, '/', '');
+                return new RegExp(StringPrototypeTrim(text));
```

Comment:
This seems like a ReDOS CVE waiting to happen - can we use search or globs here instead of regex?

Reply:
I wasn't convinced on the regex so for now I've reverted back to what I had before without it.